development/tools/jobs/incubator/reconciler/reconciler_test.go

```diff
@@ -95,7 +95,9 @@ func TestReconcilerJobsPeriodicE2EUpgrade(t *testing.T) {
 	// THEN
 	require.NoError(t, err)
 
+	assert.Len(t, jobConfig.Periodics, 3)
```

Comment:
Why do you need such assertion for the number of jobs? Each test is checking the existence of a given job. The additional assertion makes the tests harder to maintain - each time you add/remove a job you need to modify multiple tests.

Reply:
They came with the revert commit. I totally agree that we don't need them, so I've deleted them from all the reconciler job tests.